docs/project-model/project-json-reference.md

```diff
@@ -0,0 +1,494 @@
+## namedResource
+Type: Globbing String
+
+an object whos keys are names of named resources and whos values are the paths to each resource.
+
+<a name="packInclude"></a>
+## packInclude
+Type: Globbing String
+
+A globbing string that specifies what files are included when creating the nuget package.
+
+<a name="exclude"></a>
+## exclude
+Type: Globbing String
+
+A globbing string that specifies what files to exclude from the build.
+
+<a name="contentBuiltIn"></a>
+## contentBuiltIn
+Type: Globbing String
+
+A globbing string that specifies what files are considerd content by the build.
+
+<a name="compileBuiltIn"></a>
+## compileBuiltIn
+Type: Globbing String
+
+A globbing string that specifies what files are included in compilation.
+
+<a name="resourceBuiltIn"></a>
+## resourceBuiltIn
+Type: Globbing String
+
+A globbing string that specifies what files are resource files.
+
+<a name="excludeBuiltIn"></a>
+## excludeBuiltIn
+Type: Globbing String
+
+A globbing string that specifies what files to exclude from the build.
+
+<a name="dependencies"></a>
+## dependencies
+Type: Object
+
+An object that defines the package dependencies of the project, each key of this object is the name of a package and each value contains versioning information.
+
+    "dependencies": {
+        "System.Reflection.Metadata": "1.2.0-rc3-23811",
+        "Microsoft.Extensions.JsonParser.Sources": {
+          "type": "build",
+          "version": "1.0.0-rc2-16453"
+        },
+        "Microsoft.Extensions.HashCodeCombiner.Sources": {
+          "type": "build",
+          "version": "1.0.0-rc2-16054"
+        },
+        "Microsoft.Extensions.DependencyModel": "1.0.0-*"
+    }
+
+<a name="tools"></a>
+## tools
+Type: Object
+
+An object that defines package dependencies that are used as tools for the current project, not as references. Packages defined here are available in scripts that run during the build process, but they are not accessible to the code in the project itself. Tools can for exampel include code generators or post build tools that perform tasks related to packing.
+
+<a name="commands"></a>
+## commands
+Type: Object
+
+Commands are depricated in the cli. (?)
+
+<a name="scripts"></a>
+## scripts
+Type: Object
+
+An object that defines scripts run during the build proces. each key in this object identifies where in the build the script is run, each value is either a string with the script to run or an array of strings containg scripts that will run in order.
+The supported events are:
+* precompile
+* postcompile
+* prepublish
+* postpublish
+
+<a name="compilationOptions"></a>
+## compilationOptions
+Type: Object
+
+An object whose properties control various aspects of compilation, the valid properties are listed below.
+
+    "compilationOptions": {
+      "allowUnsafe": true
+    }
+
+<a name="define"></a>
+### define
+Type: String[]
+
+A list of defines such as "DEBUG" or "TRACE" that can be used in conditional compilation in the code.
```

Comment:
Do you mean like "This can also be used to set target specific defines[...]" or "The target framework also gets its own define automatically"? i'm not quite sure what the behavior is but i think there is already an automatic define for each target.. i'll look into that though